### PR TITLE
LPS-78741

### PR DESCRIPTION
**Pull request by petershin**
Hi Andrea!

It looks like the version of `npm` we're using doesn't support the `ci` command.

I think `node` installs `npm` so I tried using the very latest `node` release, but its still using an old `npm` release.

Do you know if its safe to update `npm` separately from `node`?

### CODE REVIEW

File: modules/sdk/gradle-plugins-node/src/main/java/com/liferay/gradle/plugins/node/tasks/NpmInstallTask.java [hunk 1]
```diff
@@ -516,5 +529,6 @@ public boolean accept(Path path) throws IOException {
 	private boolean _nodeModulesCacheNativeSync = true;
 	private Object _nodeModulesDigestFile;
 	private Object _removeShrinkwrappedUrls;
+	private Object _useNpmCI = false;
```
Comment (Ithildir) [inline modules/sdk/gradle-plugins-node/src/main/java/com/liferay/gradle/plugins/node/tasks/NpmInstallTask.java]: If the default value is `false`, you can omit it.

File: modules/sdk/gradle-plugins-node/README.markdown [hunk 2]
```diff
@@ -210,6 +210,7 @@ Property Name | Type | Default Value | Description
...
 `nodeModulesDigestFile` | `File` | `null`. | <p>If this property is set, the content of the project's `package-lock.json` (or `npm-shrinkwrap.json`, or `package.json` if absent) is checked with the digest from the `node_modules` directory. If the digests don't match, the `node_modules` directory is deleted before running `npm install`.</p><p>This feature is not available if the `com.liferay.cache` plugin is applied, or if the property `nodeModulesCacheDir` is set.</p>
 `removeShrinkwrappedUrls` | `boolean` | `true` if the [registry](#registry) property has a value, `false` otherwise. | Whether to temporarily remove all the hard-coded URLs in the `from` and `resolved` fields of the `npm-shinkwrap.json` file before invoking `npm install`. This way, it is possible to force NPM to download all dependencies from a custom registry declared in the [`registry`](#registry) property.
+`useNpmCI` | `boolean` | `false` | If `true`, run `npm ci` instead of `npm install`.
```
Comment (Ithildir) [inline modules/sdk/gradle-plugins-node/README.markdown]: I'd say:

> Whether to run [`npm ci`](https://docs.npmjs.com/cli/ci) instead of `npm install`.

Also, we should say that the `nodeModulesDigestFile` property has no effect if `useNpmCI` is `true`.

**Comment by petershin** [comment]
Will apply the changes.  I'm not sure if you saw my earlier comment.

The version of `npm` that we're using doesn't support the `ci` command.  Do you know the best way to update `npm` to use the latest version (`5.7.1`)?

**Comment by Ithildir** [comment]
Sorry, I missed it 😞 @jbalsas, what version of Node.js should we use to have `npm ci`?

**Comment by jbalsas** [comment]
Hey @Ithildir, @petershin, `ci` was released in [npm@5.7.0](https://github.com/npm/npm/releases/tag/v5.7.0), although this one introduced a seriour permission error, so it's better to jump straight to [npm@5.7.1](https://github.com/npm/npm/releases/tag/v5.7.1)

As per https://github.com/nodejs/Release#release-plan, a new major version of node won't be released until September 2018

As far as I can tell, no version of node has been released already bundling these versions, so we should probably just get us up to date with:
- [node@8.10.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V8.md#2018-03-06-version-8100-carbon-lts-gibfahn)
- [npm@5.7.1](https://github.com/npm/npm/releases/tag/v5.7.1)

**Comment by petershin** [comment]
Ran a few tests on the [flags-taglib](https://github.com/liferay/liferay-portal/tree/master/modules/apps/collaboration/flags/flags-taglib) submodule and `install` was a few seconds faster than `ci`.

I'm running `ant all` on my laptop to see if there's a noticeable difference there

**Comment by petershin** [comment]
I ran one test for each b/c it takes so long to run `ant all` :smile:

Running `ci`: 35 minutes 4 seconds
Running `install`: 39 minutes 50 seconds

**Comment by jbalsas** [comment]
Not impressed 😂 

I'll try tomorrow from the office too, to compare different runs!

**Comment by jbalsas** [comment]
I certainly didn't see any speed improvements, but at least the `lock` files weren't updated 🤷‍♂️

**Comment by Ithildir** [comment]
So @jbalsas, should we go ahead and merge it anyway?

**Comment by jbalsas** [comment]
I guess?

File: modules/sdk/gradle-plugins-node/README.markdown [hunk 3]
```diff
@@ -208,8 +208,9 @@ Property Name | Type | Default Value | Description
...
+`nodeModulesDigestFile` | `File` | `null`. | <p>If this property is set, the content of the project's `package-lock.json` (or `npm-shrinkwrap.json`, or `package.json` if absent) is checked with the digest from the `node_modules` directory. If the digests don't match, the `node_modules` directory is deleted before running `npm install`.</p><p>This feature is not available if the `com.liferay.cache` plugin is applied, if the property `nodeModulesCacheDir` is set, or if the property `useNpmCI` is set to `true`.</p>
 `removeShrinkwrappedUrls` | `boolean` | `true` if the [registry](#registry) property has a value, `false` otherwise. | Whether to temporarily remove all the hard-coded URLs in the `from` and `resolved` fields of the `npm-shinkwrap.json` file before invoking `npm install`. This way, it is possible to force NPM to download all dependencies from a custom registry declared in the [`registry`](#registry) property.
+`useNpmCI` | `boolean` | `false` | If `true`, run `npm ci` instead of `npm install`.
```
Comment (Ithildir) [inline modules/sdk/gradle-plugins-node/README.markdown]: Can you please rephrase it to:
> Whether to run `npm ci` instead of `npm install`

Thanks!

**Comment by petershin** [comment]
Appreciate the review! Added the changes

**Comment by jbalsas** [comment]
Hey @Ithildir, @petershin, what's the status of this?

We're currently seeing some test errors caused by some newly published modules that aren't compatible with the current code. These type of issues should be avoided if we use `npm ci` as well since it should always respect the `package-lock` contents.

**Comment by Ithildir** [comment]
Hi @jbalsas,
I had to work on the 7.1.x fork, I'll try to forward this pull today!

**Comment by jbalsas** [comment]
No worries, no need to rush!! ☺️

**Comment by Ithildir** [comment]
Sent to https://github.com/brianchandotcom/liferay-portal/pull/56245.
